### PR TITLE
Create law and test: energy from melting

### PR DESCRIPTION
**Pull request by Voldoned**

### CODE REVIEW

File: symplyphysics/laws/thermodynamics/energy_from_melting.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+from symplyphysics import (units, Quantity, Symbol, print_expression, validate_input,
+    validate_output)
+
+# Description
+## The heat of melting is the amount of heat that must be brought to a solid
+# crystalline substance at constant pressure in order to completely transfer it
+# to a liquid state.
+## Q = k_lambda * m
+## Where:
+## Q = energy from melting
```
Comment (blackyblack) [inline symplyphysics/laws/thermodynamics/energy_from_melting.py:11]: It's not energy from melting. It's energy, required to melt something.
Also rename the law file name.

Comment (Voldoned) [reply symplyphysics/laws/thermodynamics/energy_from_melting.py:11]: Is the name "energy_to_melt_solid_matter" appropriate?

Comment (blackyblack) [reply symplyphysics/laws/thermodynamics/energy_from_melting.py:11]: For file name? I think `energy_to_melt_from_mass`

Comment (Voldoned) [reply symplyphysics/laws/thermodynamics/energy_from_melting.py:11]: > Для имени файла?

Yes

File: test/thermodynamics/energy_to_melt_from_mass_test.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+from pytest import approx, fixture, raises
+from symplyphysics import (
+    errors,
+    units,
+    Quantity,
+    SI,
+    convert_to,
+    prefixes,
+)
+from symplyphysics.laws.thermodynamics import energy_from_combustion as amount_energy
```
Comment (blackyblack) [inline test/thermodynamics/energy_to_melt_from_mass_test.py:11]: How come you're testing another law, yet there is no any problem in tests?

File: symplyphysics/laws/thermodynamics/energy_to_melt_from_mass.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+## Where:
+## Q = energy, required to melt solid matter
+## k_lambda - specific heat of melting
+## m - mass of solid matter
+
+amount_energy = Symbol("amount_energy", units.energy)
+specific_heat_melting = Symbol("specific_heat_melting", units.energy / units.mass)
+mass_of_matter = Symbol("mass_of_matter", units.mass)
+
+law = Eq(amount_energy, specific_heat_melting * mass_of_matter)
```
Comment (blackyblack) [inline symplyphysics/laws/thermodynamics/energy_to_melt_from_mass.py:19]: So this law is exactly the same as https://github.com/blackyblack/symplyphysics/blob/master/symplyphysics/laws/thermodynamics/energy_from_combustion.py
We probably should keep it but link somehow to another law.

Comment (Voldoned) [reply symplyphysics/laws/thermodynamics/energy_to_melt_from_mass.py:19]: The formulas of these laws are very similar, only the meaning of the specific heat constants changes.

Comment (blackyblack) [reply symplyphysics/laws/thermodynamics/energy_to_melt_from_mass.py:19]: There is no perfect way to solve it. I suggest to mention in the description about the similarities between these laws. And also put some comment to hint user to clearly distinguish specific heat of combustion, of melting or any other specific heats.

Comment (Voldoned) [reply symplyphysics/laws/thermodynamics/energy_to_melt_from_mass.py:19]: Thank you very much!

File: symplyphysics/laws/thermodynamics/energy_to_melt_from_mass.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+amount_energy = Symbol("amount_energy", units.energy)
+specific_heat_melting = Symbol("specific_heat_melting", units.energy / units.mass)
+mass_of_matter = Symbol("mass_of_matter", units.mass)
+
+# Important note! Although we substitute the specific heat of melting
+# into the specific heat of combustion in the formula of the thermal energy
+# released during the combustion of a substance, these values should be distinguished,
+# they are not equivalent!
+energy_from_combustion_value = combustion_energy_law.law.subs({
```
Comment (blackyblack) [inline symplyphysics/laws/thermodynamics/energy_to_melt_from_mass.py:25]: Don't do it. Write in the description about similarities. Also mention specific heat difference near the specific heat usage.

Comment (Voldoned) [reply symplyphysics/laws/thermodynamics/energy_to_melt_from_mass.py:27]: Like in commit 07c9bb6 ?

Comment (blackyblack) [reply symplyphysics/laws/thermodynamics/energy_to_melt_from_mass.py:27]: Don't reuse another law.

File: symplyphysics/laws/thermodynamics/energy_to_melt_from_mass.py [hunk 5]
```diff
@@ -2,34 +2,31 @@
...
 ## to one unit of the mass of a crystalline substance in an equilibrium
 ## isobaric-isothermal process in order to transfer it from a solid (crystalline)
-##state to a liquid one
-#
-# NOTICE: specific heat of melting and specific heat of combustion values should
-## be distinguished, they are not equivalent!
-## Specific heat of combustion - a physical quantity showing how much heat is released when a fuel weighing 1 kg is completely burned.
+## state to a liquid one
+##
+## NOTICE: This law is similar to the law of energy released during
```
Comment (blackyblack) [inline symplyphysics/laws/thermodynamics/energy_to_melt_from_mass.py:17]: You can put a similar notice in combustion law.

**Review by blackyblack — APPROVED** [review]

**Pull request merged**